src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -0,0 +1,342 @@
+        size_t index = 0;
+
+        VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+
+        // operational state list is not found
+        if (delegate->GetOperationalStateAtIndex(index, opState) == CHIP_ERROR_NOT_FOUND)
+        {
+            err = aEncoder.EncodeNull();
+        }
+        else
+        {
+            return aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
+                while (delegate->GetOperationalStateAtIndex(index, opState) != CHIP_ERROR_NOT_FOUND)
+                {
+                    err = encoder.Encode(opState);
+                    if (err != CHIP_NO_ERROR)
+                    {
+                        return err;
+                    }
+                    index++;
+                }
+                return CHIP_NO_ERROR;
+            });
+        }
+    }
+    break;
+
+    case OperationalState::Attributes::OperationalState::Id: {
+
+        Delegate * delegate = OperationalState::GetOperationalStateDelegate(mEndpointId, mClusterId);
+        GenericOperationalState opState(to_underlying(OperationalStateEnum::kStopped));
+        VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+        delegate->GetOperationalState(opState);
+        return aEncoder.Encode(opState);
+    }
+    break;
+
+    case OperationalState::Attributes::OperationalError::Id: {
+        Delegate * delegate = OperationalState::GetOperationalStateDelegate(mEndpointId, mClusterId);
+        GenericOperationalError opErr(to_underlying(ErrorStateEnum::kNoError));
+        VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+        delegate->GetOperationalError(opErr);
+        return aEncoder.Encode(opErr);
+    }
+    break;
+
+    case OperationalState::Attributes::PhaseList::Id: {
+        Delegate * delegate = OperationalState::GetOperationalStateDelegate(mEndpointId, mClusterId);
+
+        GenericOperationalPhase phase = GenericOperationalPhase(DataModel::Nullable<CharSpan>());
+        size_t index                  = 0;
+
+        VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+        err = delegate->GetOperationalPhaseAtIndex(index, phase);
+        if (err == CHIP_ERROR_NOT_FOUND || phase.isNullable())
+        {
+            err = aEncoder.EncodeNull();
+        }
+        else
+        {
+            return aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
+                while (delegate->GetOperationalPhaseAtIndex(index, phase) != CHIP_ERROR_NOT_FOUND)
+                {
+                    err = encoder.Encode(phase.mPhaseName);
+                    if (err != CHIP_NO_ERROR)
+                    {
+                        return err;
+                    }
+                    index++;
+                }
+                return CHIP_NO_ERROR;
+            });
+        }
+    }
+    break;
+
+    case OperationalState::Attributes::CurrentPhase::Id: {
```

Comment:
Please add this one to the set of AttributeAccessInterface-only attributes in src/app/zap-templates/zcl/zcl.json and src/app/zap-templates/zcl/zcl-with-test-extensions.json.